packages/algob/test/lib/tx.ts

```diff
@@ -22,3 +35,72 @@ describe("Note in TxParams", () => {
     assert.deepEqual(result, encoder.encode(noteb64), "noteb64 not encoded");
   });
 });
+
+describe("Opt-In to ASA", () => {
+  function mkASA (): types.ASADef {
+    return {
+      total: 1,
+      decimals: 1,
+      unitName: 'ASA',
+      defaultFrozen: false
+    };
+  }
+
+  let deployer: Deployer;
+  let execParams: types.OptInASAParam;
+  let algod: AlgoOperatorDryRunImpl;
+  let fn: any;
+  let expected: ConfirmedTxInfo;
+  beforeEach(async () => {
+    const env = mkEnv("network1");
+    algod = new AlgoOperatorDryRunImpl();
+    const deployerCfg = new DeployerConfig(env, algod);
+    deployerCfg.asaDefs = { silver: mkASA() };
+    deployer = new DeployerDeployMode(deployerCfg);
+    await deployer.deployASA("silver", { creator: deployer.accounts[0] });
+    execParams = {
+      type: types.TransactionType.OptInASA,
+      sign: types.SignType.SecretKey,
+      payFlags: {},
+      fromAccount: bobAcc,
+      assetID: 1
+    };
+    fn = sinon.stub(algod.algodClient, "getTransactionParams")
+      .returns({ do: async () => mockSuggestedParam });
+    expected = {
+      'confirmed-round': 1,
+      "asset-index": 1,
+      'application-index': 1,
+      'global-state-delta': "string",
+      'local-state-delta': "string"
+    };
+  });
+
+  afterEach(() => {
+    fn.restore();
```

Comment:
```suggestion
    algod.algodClient.restore();
```
an you can remove fn assignment in line 68 and declaration in line 52

Reply:
i don't think we can do this `Property 'restore' does not exist on type 'Algodv2'`

Reply:
hmm, it should work:
```
> sinon.stub(console, 'warn')    // NOTE - we don't to store the returned value
> console.warn("abc")            // NOTE - nothing is printed
> console.warn.restore()
> console.warn("abc")   // after restore we see printed "abc"
abc
```